File: java/src/test/java/com/example/app/MainTest.java
package com.example.app;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MainTest {

    @Test
    void greetReturnsExpectedMessage() {
        assertEquals("Hello, Polyglot-App-Template!", Main.greet("Polyglot-App-Template"));
    }

    @Test
    void greetHandlesDefaultCase() {
        assertEquals("Hello, world!", Main.greet("world"));
    }
}

File: kotlin/src/test/kotlin/com/example/app/MainTest.kt
package com.example.app

import kotlin.test.Test
import kotlin.test.assertEquals

class MainTest {
    @Test
    fun `greet returns expected message`() {
        assertEquals("Hello, Polyglot-App-Template!", greet("Polyglot-App-Template"))
    }

    @Test
    fun `greet handles default case`() {
        assertEquals("Hello, world!", greet("world"))
    }
}
